core/storage/xen.py

```diff
@@ -249,3 +270,67 @@ def prepare_for_vm_startup(self, verbose):
                     raise QubesException(
                         "VM '{}' holding '{}' does not exists".format(
                             drive_domain, drive_path))
+
+
+class XenPool(Pool):
+
+    def __init__(self, vm, dir):
+        assert vm is not None
+        assert dir is not None
+
+        appvms_path = os.path.join(dir, 'appvms')
+        servicevms_path = os.path.join(dir, 'servicevms')
+        vm_templates_path = os.path.join(dir, 'vm-templates')
+
+        self._create_dir_if_not_exists(dir)
+        self._create_dir_if_not_exists(appvms_path)
+        self._create_dir_if_not_exists(servicevms_path)
+        self._create_dir_if_not_exists(vm_templates_path)
+
+        self.vmdir = self._vmdir_path(vm, dir)
+        self.vm = vm
+        self.dir = dir
+
+    def getStorage(self):
+        """ Returns an instantiated ``XenStorage``. """
+        return defaults['storage_class'](self.vm, vmdir=self.vmdir)
+
+    def _vmdir_path(self, vm, pool_dir):
+        """ Get the vm dir depending on the type of the VM.
+
+            The default QubesOS file storage saves the vm images in three
+            different directories depending on the ``QubesVM`` type:
+
+            * ``appvms`` for ``QubesAppVm`` or ``QubesHvm``
+            * ``vm-templates`` for ``QubesTemplateVm`` or ``QubesTemplateHvm``
+            * ``servicevms`` for ``QubesProxyVm``, ``QubesNetVm`` or
+                ``QubesAdminVm``
+
+            Args:
+                vm: a QubesVM
+                pool_dir: the root directory of the pool
+
+            Returns:
+                string (str) absolute path to the directory where the vm files
+                             are stored
+        """
+        vm_type = type(vm)
+
+        if vm_type in [QubesAppVm, QubesHVm]:
+            subdir = 'appvms'
+        elif vm_type in [QubesTemplateVm, QubesTemplateHVm]:
+            subdir = 'vm-templates'
+        elif vm_type in [QubesAdminVm, QubesNetVm, QubesProxyVm]:
+            subdir = 'servicevms'
+        else:
+            raise QubesException(str(vm_type) + ' unknown vm type')
```

Comment:
Not sure about this one. Maybe better default to some value? 'appvms'? 'other'?
How would that work for example with QubesDisposableVm?